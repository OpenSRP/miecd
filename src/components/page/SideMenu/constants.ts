import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { NavigationModule, PageLink } from './SubMenu';
import { TFunction } from 'react-i18next';
import {
    ANC_URL,
    CHILD_URL,
    CLIENT_URL,
    HOME_URL,
    HOUSEHOLD_URL,
    LOCATIONS_URL,
    NBC_AND_PNC_ANALYSIS_URL,
    NBC_AND_PNC_COMPARTMENTS_URL,
    NBC_AND_PNC_LOGFACE_URL,
    NBC_AND_PNC_URL,
    NUTRITION_ANALYSIS_URL,
    NUTRITION_COMPARTMENTS_URL,
    NUTRITION_LOGFACE_URL,
    NUTRITION_URL,
    PREGNANCY_ANALYSIS_URL,
    PREGNANCY_COMPARTMENTS_URL,
    PREGNANCY_LOGFACE_URL,
    PREGNANCY_URL,
    REPORTS_URL,
    ROLE_URL,
    TEAM_URL,
    USER_URL,
} from '../../../constants';

import {
    ENABLE_ANC_PAGE,
    ENABLE_CHILD_PAGE,
    ENABLE_CLIENT_PAGE,
    ENABLE_HOUSEHOLD_PAGE,
    ENABLE_LOCATION_PAGE,
    ENABLE_ROLE_PAGE,
    ENABLE_TEAM_PAGE,
    ENABLE_USER_PAGE,
} from '../../../configs/env';
import { Dictionary } from '@onaio/utils';

export const navigationPageLinks = (t: TFunction): Dictionary<PageLink> => {
    return {
        // Page links
        CLIENT_PAGE_NAVIGATION: {
            label: t('All clients'),
            url: CLIENT_URL,
        },
        HOUSEHOLD_PAGE_NAVIGATION: {
            label: t('Household'),
            url: HOUSEHOLD_URL,
        },
        ANC_PAGE_NAVIGATION: {
            label: t('ANC'),
            url: ANC_URL,
        },
        CHILD_PAGE_NAVIGATION: {
            label: t('Child'),
            url: CHILD_URL,
        },

        USERS_PAGE_NAVIGATION: {
            label: t('Users'),
            url: USER_URL,
        },
        ROLES_PAGE_NAVIGATION: {
            label: t('Roles'),
            url: ROLE_URL,
        },
        TEAMS_PAGE_NAVIGATION: {
            label: t('Teams'),
            url: TEAM_URL,
        },
        LOCATIONS_PAGE_NAVIGATION: {
            label: t('Locations'),
            url: LOCATIONS_URL,
        },

        PREGNANCY_PAGE_NAVIGATION: {
            label: t('Pregnancy'),
            url: PREGNANCY_URL,
        },

        NUTRITION_PAGE_NAVIGATION: {
            label: t('Nutrition'),
            url: NUTRITION_URL,
        },

        NEWBORN_AND_POSTNATAL_PAGE_NAVIGATION: {
            label: t('NBC & PNC'),
            url: NBC_AND_PNC_URL,
        },

        PREGNANCY_COMPARTMENTS_PAGE_NAVIGATION: {
            label: t('Compartments'),
            url: PREGNANCY_COMPARTMENTS_URL,
        },

        NBC_AND_PNC_COMPARTMENTS_PAGE_NAVIGATION: {
            label: t('Compartments'),
            url: NBC_AND_PNC_COMPARTMENTS_URL,
        },

        NUTRITION_COMPARTMENTS_PAGE_NAVIGATION: {
            label: t('Compartments'),
            url: NUTRITION_COMPARTMENTS_URL,
        },

        NBC_AND_PNC_LOGFACE_PAGE_NAVIGATION: {
            label: t('Log face'),
            url: NBC_AND_PNC_LOGFACE_URL,
        },

        PREGNANCY_LOGFACE_PAGE_NAVIGATION: {
            label: t('Log face'),
            url: PREGNANCY_LOGFACE_URL,
        },

        NUTRITION_LOGFACE_PAGE_NAVIGATION: {
            label: t('Log face'),
            url: NUTRITION_LOGFACE_URL,
        },

        PREGNANCY_ANALYSIS_PAGE_NAVIGATION: {
            label: t('Analysis'),
            url: PREGNANCY_ANALYSIS_URL,
        },

        NBC_AND_PNC_ANALYSIS_PAGE_NAVIGATION: {
            label: t('Analysis'),
            url: NBC_AND_PNC_ANALYSIS_URL,
        },

        NUTRITION_ANALYSIS_PAGE_NAVIGATION: {
            label: t('Analysis'),
            url: NUTRITION_ANALYSIS_URL,
        },
    };
};

// icons
export const homeNavIcon: IconProp = ['fas', 'home'];
export const pregancyNavIcon: IconProp = ['far', 'user'];
export const nbcandpncNavIcon: IconProp = ['far', 'user'];
export const nutritionNavIcon: IconProp = ['far', 'user'];
export const newbornAndPostnatalNavIcon: IconProp = ['far', 'user'];
export const clientModuleNavIcon: IconProp = ['far', 'user'];
export const reportModuleNavIcon: IconProp = ['fas', 'chart-line'];
export const adminModuleNavIcon: IconProp = ['fas', 'cog'];

// Navigation links for navigation module. collapses when clicked to display the pageLinks
export const modulePageLinks = (t: TFunction) => {
    const links = {
        HOME_PARENT_NAV: {
            icon: homeNavIcon,
            label: t('Home'),
            url: HOME_URL,
        },
        NBC_AND_PNC_PARENT_NAV: {
            icon: nbcandpncNavIcon,
            label: t('NBC & PNC'),
            url: NBC_AND_PNC_URL,
        },
        NUTRITION_PARENT_NAVIGATION: {
            icon: nutritionNavIcon,
            label: t('Nutrition'),
            url: NUTRITION_URL,
        },
        PREGNANCY_MODULE_PARENT_NAV: {
            icon: pregancyNavIcon,
            label: t('Pregnancy'),
            url: PREGNANCY_URL,
        },
        CLIENT_MODULE_PARENT_NAV: {
            icon: clientModuleNavIcon,
            label: t('Client Records'),
        },
        REPORT_MODULE_PARENT_NAV: {
            icon: reportModuleNavIcon,
            label: t('Reports'),
            url: REPORTS_URL,
        },
        ADMIN_MODULE_PARENT_NAV: {
            icon: adminModuleNavIcon,
            label: t('Admin'),
        },
    };
    return links;
};

export const navigationModulesFactory = (t: TFunction) => {
    const {
        PREGNANCY_MODULE_PARENT_NAV,
        NBC_AND_PNC_PARENT_NAV,
        NUTRITION_PARENT_NAVIGATION,
        CLIENT_MODULE_PARENT_NAV,
        HOME_PARENT_NAV,
        ADMIN_MODULE_PARENT_NAV,
        REPORT_MODULE_PARENT_NAV,
    } = modulePageLinks(t);
    const {
        PREGNANCY_COMPARTMENTS_PAGE_NAVIGATION,
        PREGNANCY_ANALYSIS_PAGE_NAVIGATION,
        PREGNANCY_LOGFACE_PAGE_NAVIGATION,
        NBC_AND_PNC_LOGFACE_PAGE_NAVIGATION,
        NBC_AND_PNC_COMPARTMENTS_PAGE_NAVIGATION,
        NBC_AND_PNC_ANALYSIS_PAGE_NAVIGATION,
        NUTRITION_LOGFACE_PAGE_NAVIGATION,
        NUTRITION_COMPARTMENTS_PAGE_NAVIGATION,
        NUTRITION_ANALYSIS_PAGE_NAVIGATION,
        CLIENT_PAGE_NAVIGATION,
        HOUSEHOLD_PAGE_NAVIGATION,
        ANC_PAGE_NAVIGATION,
        CHILD_PAGE_NAVIGATION,
        USERS_PAGE_NAVIGATION,
        ROLES_PAGE_NAVIGATION,
        TEAMS_PAGE_NAVIGATION,
        LOCATIONS_PAGE_NAVIGATION,
    } = navigationPageLinks(t);
    // nav module objects
    const navigationModules: Dictionary<NavigationModule> = {
        HOME_NAVIGATION_MODULE: {
            childNavs: [],
            parentNav: HOME_PARENT_NAV,
        },

        PREGNANCY_NAVIGATION_MODULE: {
            childNavs: [
                PREGNANCY_LOGFACE_PAGE_NAVIGATION,
                PREGNANCY_COMPARTMENTS_PAGE_NAVIGATION,
                PREGNANCY_ANALYSIS_PAGE_NAVIGATION,
            ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
            parentNav: PREGNANCY_MODULE_PARENT_NAV,
        },

        NBC_AND_PNC_NAVIGATION_MODULE: {
            childNavs: [
                NBC_AND_PNC_LOGFACE_PAGE_NAVIGATION,
                NBC_AND_PNC_COMPARTMENTS_PAGE_NAVIGATION,
                NBC_AND_PNC_ANALYSIS_PAGE_NAVIGATION,
            ],
            parentNav: NBC_AND_PNC_PARENT_NAV,
        },

        NUTRITION_MODULE: {
            childNavs: [
                NUTRITION_LOGFACE_PAGE_NAVIGATION,
                NUTRITION_COMPARTMENTS_PAGE_NAVIGATION,
                NUTRITION_ANALYSIS_PAGE_NAVIGATION,
            ],
            parentNav: NUTRITION_PARENT_NAVIGATION,
        },

        CLIENT_NAVIGATION_MODULE: {
            childNavs: [
                ENABLE_CLIENT_PAGE && CLIENT_PAGE_NAVIGATION,
                ENABLE_HOUSEHOLD_PAGE && HOUSEHOLD_PAGE_NAVIGATION,
                ENABLE_ANC_PAGE && ANC_PAGE_NAVIGATION,
                ENABLE_CHILD_PAGE && CHILD_PAGE_NAVIGATION,
            ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
            parentNav: CLIENT_MODULE_PARENT_NAV,
        },
        REPORT_NAVIGATION_MODULE: {
            childNavs: [],
            parentNav: REPORT_MODULE_PARENT_NAV,
        },
        ADMIN_NAVIGATION_MODULE: {
            childNavs: [
                ENABLE_USER_PAGE && USERS_PAGE_NAVIGATION,
                ENABLE_ROLE_PAGE && ROLES_PAGE_NAVIGATION,
                ENABLE_TEAM_PAGE && TEAMS_PAGE_NAVIGATION,
                ENABLE_LOCATION_PAGE && LOCATIONS_PAGE_NAVIGATION,
            ].filter<PageLink>((childNav: PageLink | boolean): childNav is PageLink => typeof childNav !== 'boolean'),
            parentNav: ADMIN_MODULE_PARENT_NAV,
        },
    };
    return navigationModules;
};

/** Enable Clients Module from its child pages */
export const ENABLE_CLIENT_RECORDS_MODULE =
    ENABLE_ANC_PAGE || ENABLE_CLIENT_PAGE || ENABLE_HOUSEHOLD_PAGE || ENABLE_HOUSEHOLD_PAGE;
export type ENABLE_CLIENT_RECORDS_MODULE = typeof ENABLE_CLIENT_RECORDS_MODULE;

/** Enable Admin Module from its child pages */
export const ENABLE_ADMIN_MODULE = ENABLE_USER_PAGE || ENABLE_ROLE_PAGE || ENABLE_TEAM_PAGE || ENABLE_LOCATION_PAGE;
export type ENABLE_ADMIN_MODULE = typeof ENABLE_ADMIN_MODULE;
