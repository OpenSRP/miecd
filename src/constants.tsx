import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { TFunction } from 'i18next';

// openSRP api endpoints
export const OPENSRP_SECURITY_AUTHENTICATE = '/security/authenticate';

// internal urls
export const CHILD_PATIENT_DETAIL = 'child_patient_detail';
export const PATIENT_DETAIL = 'patient_detail';
export const NBC_AND_PNC_URL = '/nbc_and_pnc';
export const HIERARCHICAL_DATA_URL = '/hierarchicaldata';
export const PREGNANCY_URL = '/pregnancy';
export const LOGIN_URL = '/login';
export const LOGOUT_URL = '/logout';
export const HOME_URL = '/';
export const PREGNANCY_COMPARTMENTS_URL = '/pregnancy_compartments';
export const NBC_AND_PNC_COMPARTMENTS_URL = '/nbc_and_pnc_compartments';
export const CLIENT_URL = '/clients';
export const LOGFACE_URL = '/log-face';
export const PREGNANCY_LOGFACE_URL = '/pregnancy_log_face';
export const NBC_AND_PNC_LOGFACE_URL = '/nbc_and_pnc_logface_url';
export const NUTRITION_LOGFACE_URL = '/nutrition_logface_url';
export const NUTRITION_COMPARTMENTS_URL = '/nutrition_commpartments_url';
export const PREGNANCY_ANALYSIS_URL = '/pregnancy_analysis';
export const NBC_AND_PNC_ANALYSIS_URL = '/nbc_and_pnc_analysis';
export const NUTRITION_ANALYSIS_URL = '/nutrition_analysis';
export const NUTRITION_URL = '/nutrition';
export const HOUSEHOLD_URL = '/404';
export const ANC_URL = '/404';
export const CHILD_URL = '/404';
export const USER_URL = '/404';
export const ROLE_URL = '/404';
export const TEAM_URL = '/404';
export const LOCATIONS_URL = '/404';
export const REPORTS_URL = '/';
export const PATIENT_DETAIL_URL = `/${PATIENT_DETAIL}/:patient_id`;
export const CHILD_PATIENT_DETAIL_URL = `/${CHILD_PATIENT_DETAIL}/:patient_id`;
export const VIETNAM_COUNTRY_LOCATION_ID = 'd1865325-11e6-4e39-817b-e676c1affecf';
export const BACKEND_CALLBACK_URL = '/fe/oauth/callback/opensrp';
export const BACKEND_CALLBACK_PATH = '/fe/oauth/callback/:id';
export const REACT_CALLBACK_PATH = '/oauth/callback/:id';
export const BACKEND_LOGIN_URL = '/fe/login';
export const EXPRESS_LOGIN_URL = '/login';
export const REACT_LOGIN_URL = '/login';
export const EXPRESS_TOKEN_REFRESH_URL = '/refresh/token';

// string literals
export const HIGH = 'high';
export const LOW = 'low';
export const RED = 'red';
export const NO = 'no';
export const OVERWEIGHT = 'overweight';
export const STUNTED = 'stunted';
export const SEVERE_WASTING = 'severe wasting';
export const INAPPROPRIATELY_FED = 'inappropriately fed';
export const NOT_SET_LOWERCASE = 'not set';
export const PREGNANCY = 'Pregnancy';
export const NBC_AND_PNC = 'NBC & PNC';
export const NBC_AND_PNC_CHILD = 'NBC & PNC_CHILD';
export const NBC_AND_PNC_WOMAN = 'NBC & PNC_WOMAN';
export const NUTRITION = 'Nutrition';
export const NO_UNDERSCORE_RISK = 'no_risk';
export const RED_ALERT = 'red_alert';
export const NORMAL = 'normal';

// module string literals
export const PREGNANCY_MODULE = 'Pregnancy';
export const NUTRITION_MODULE = 'Nutrition';
export const NBC_AND_PNC_MODULE = 'NBC & PNC';

export const ALL = 'all';
export const UP = 'up';
export const NO_RISK_LOWERCASE = 'no risk';
export const NO_RISK = 'No Risk';
export const VIETNAM = 'Vietnam';
export const COUNTRY = 'Country';
export const PROVINCE = 'Province';
export const COMMUNE = 'Commune';
export const DISTRICT = 'District';
export const VILLAGE = 'Village';
export const RED_ALERT_CLASSNAME = 'red-alert';
export const NEWBORN_REPORT = 'Newborn Report';
export const EC_WOMAN = 'ec_woman';
export const EC_FAMILY_MEMBER = 'ec_family_member';
export const EC_CHILD = 'ec_child';

// smsTypes
export const PREGNANCY_DETECTION = 'Pregnancy Detection';
export const PREGNANCY_IDENTIFICATION = 'Pregnancy Identification';
export const PREGNANCY_REGISTRATION = 'Pregnancy Registration';
export const ANC_REPORT = 'ANC Report';
export const HOME_VISIT_REPORT = 'Home Visit Report';
export const SOCIAL_DETERMINANTS = 'Social Determinants';
export const DELIVERY_PLANNING = 'Delivery Planning';
export const BIRTH_REPORT = 'Birth Report';
export const DEATH_REPORT = 'Death Report';
export const RED_ALERT_REPORT = 'Red Alert Report';
export const RESPONSE_REPORT = 'Response Report';
export const REFUSAL_REPORT = 'Refusal Report';
export const DEPARTURE_CODE = 'Departure Code';
export const POSTNATAL_AND_NEWBORN_CARE = 'Postnatal and Newborn Care';
export const NUTRITION_REGISTRATION = 'Nutrition Registration';
export const MONTHLY_NUTRITION_REPORT = 'Monthly Nutrition Report';
export const ACCOUNT_CHECK = 'Account Check';
export const ANC_VISIT = 'ANC Visit';

// // sms events fields
export const NUTRITION_STATUS = 'nutrition_status';
export const GROWTH_STATUS = 'growth_status';
export const FEEDING_CATEGORY = 'feeding_category';
export const EVENT_ID = 'event_id';
export const LOGFACE_RISK = 'logface_risk';
export const HEIGHT = 'height';
export const WEIGHT = 'weight';
export const RISK = 'risk';
export const NUTRITION_REPORT = 'Nutrition Report';
export const RISK_LEVEL = 'risk_level';

// TIME constants
export const MICROSECONDS_IN_A_WEEK = 604800000;
export const GESTATION_PERIOD = 24192000000;

// Back arrow constant
export const BACKPAGE_ICON: IconProp = ['fas', 'arrow-left'];

// monthNames
export const getMonthNames = (t: TFunction) => [
    t('January'),
    t('February'),
    t('March'),
    t('April'),
    t('May'),
    t('June'),
    t('July'),
    t('August'),
    t('September'),
    t('October'),
    t('November'),
    t('December'),
];

// operational constants
export const DEFAULT_NUMBER_OF_LOGFACE_ROWS = 5;
export const EN_LANGUAGE_CODE = 'en' as const;
export const VI_LANGUAGE_CODE = 'vi' as const;

export const ALL_LANGUAGE_OPTIONS = {
    [EN_LANGUAGE_CODE]: 'English',
    [VI_LANGUAGE_CODE]: 'Vietnamese',
};

export const DATE_FORMAT = 'dd/MM/yyyy';
export const EVENT_DATE_DATE_FORMAT = 'yyyy-MM-dd';
export const SMS_MESSAGE_DATE_DATE_FORMAT = 'dd-MM-yyyy';
export const LOCATION_FILTER_PARAM = 'locationSearch';
export const RISK_CATEGORY_FILTER_PARAM = 'riskCategory';
export const SMS_TYPE_FILTER_PARAM = 'smsType';
export const SEARCH_FILTER_PARAM = 'search';

// react-query queryKey's
// compartment slices
export const FETCH_COMPARTMENTS_PREGNANCY_SLICE = 'FETCH_COMPARTMENTS_PREGNANCY_SLICE';
export const FETCH_COMPARTMENTS_NBC_AND_PNC_SLICE = 'FETCH_COMPARTMENTS_NBC_AND_PNC_SLICE';
export const FETCH_COMPARTMENTS_NUTRITION_SLICE = 'FETCH_COMPARTMENTS_NUTRITION_SLICE';
// location slices
export const FETCH_VILLAGES = 'FETCH_VILLAGES';
export const FETCH_COMMUNES = 'FETCH_COMMUNES';
export const FETCH_DISTRICTS = 'FETCH_DISTRICTS';
export const FETCH_PROVINCES = 'FETCH_PROVINCES';
// user location slice
export const FETCH_USER_LOCATION = 'FETCH_USER_LOCATION';
// user id
export const FETCH_USER_ID = 'FETCH_USER_ID';
