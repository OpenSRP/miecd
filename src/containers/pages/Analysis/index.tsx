import * as React from 'react';
import { WithTranslation, withTranslation } from 'react-i18next';
import Ripple from '../../../components/page/Loading';
import './index.css';

interface State {
    loading: boolean;
}

interface Props {
    endpoint: string;
    module: string;
}

const defaultProps: Props = {
    endpoint: '',
    module: '',
};
export type AnalysisPropTypes = Props & WithTranslation;
export class Analysis extends React.Component<AnalysisPropTypes, State> {
    public static defaultProps = defaultProps;

    constructor(props: AnalysisPropTypes) {
        super(props);
        this.state = {
            loading: true,
        };
    }

    public render() {
        const { t } = this.props;
        return (
            <div className="analysis">
                <div>
                    <h2 className="analysis-title">{t(`${this.props.module} - Analysis`)}</h2>
                </div>
                <div className="analysis-wrapper">
                    {this.state.loading ? <Ripple /> : null}
                    <iframe
                        seamless
                        scrolling="yes"
                        frameBorder="0"
                        onLoad={this.hideSpinner}
                        src={this.props.endpoint}
                        title="pregnancy sms events analysis"
                        className="analysis-iframe"
                    />
                </div>
            </div>
        );
    }

    private hideSpinner = () => {
        this.setState({
            loading: false,
        });
    };
}

export default withTranslation()(Analysis);
