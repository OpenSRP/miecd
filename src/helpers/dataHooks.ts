import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
