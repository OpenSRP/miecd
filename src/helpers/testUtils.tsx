import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
