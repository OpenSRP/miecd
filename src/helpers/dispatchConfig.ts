import { ConfigState, setAllConfigs, getAllConfigs } from '@opensrp/pkg-config';
import { OPENSRP_REST_API_BASE_URL } from 'configs/settings';
import { BACKEND_ACTIVE } from '../configs/env';
import { BACKEND_LOGIN_URL, REACT_LOGIN_URL } from '../constants';
import i18n from '../mls';

export const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;

const defaultvalues = getAllConfigs();

const configObject: ConfigState = {
    ...defaultvalues,
    appLoginURL: APP_LOGIN_URL,
    opensrpBaseURL: OPENSRP_REST_API_BASE_URL,
    i18n,
};

setAllConfigs(configObject);
