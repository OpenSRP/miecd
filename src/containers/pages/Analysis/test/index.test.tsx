import { PREGNANCY_MODULE, NBC_AND_PNC_MODULE, NUTRITION_MODULE } from '../../../../constants';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { mountWithTranslations } from 'helpers/testUtils';
import React from 'react';
import Analysis from '..';

describe('Analysis', () => {
    // eslint-disable-next-line jest/expect-expect
    it('must render without crashing', () => {
        shallow(<Analysis module={PREGNANCY_MODULE} />);
    });

    it('must render correctly nutrition', () => {
        const wrapper = mountWithTranslations(<Analysis module={NUTRITION_MODULE} />);
        expect(wrapper.find('div.analysis h2').text()).toMatchSnapshot();
        expect(toJson(wrapper.find('div.analysis iframe'))).toMatchSnapshot('Analysis iframe');
    });

    it('must render correctly pregnancy', () => {
        const wrapper = mountWithTranslations(<Analysis module={PREGNANCY_MODULE} />);
        expect(wrapper.find('div.analysis h2').text()).toMatchSnapshot();
        expect(toJson(wrapper.find('div.analysis iframe'))).toMatchSnapshot('Analysis iframe');
    });

    it('must render correctly NBC_AND_PNC', () => {
        const wrapper = mountWithTranslations(<Analysis module={NBC_AND_PNC_MODULE} />);
        expect(wrapper.find('div.analysis h2').text()).toMatchSnapshot();
        expect(toJson(wrapper.find('div.analysis iframe'))).toMatchSnapshot('Analysis iframe');
    });
});
