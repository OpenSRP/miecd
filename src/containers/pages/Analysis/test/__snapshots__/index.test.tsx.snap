// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analysis must render correctly NBC_AND_PNC 1`] = `"NBC & PNC - Analysis"`;

exports[`Analysis must render correctly NBC_AND_PNC: Analysis iframe 1`] = `
<iframe
  className="analysis-iframe"
  frameBorder="0"
  onLoad={[Function]}
  scrolling="yes"
  seamless={true}
  src="https://somesuperseturl.org/superset/dashboard/10/?standalone=true"
  title="pregnancy sms events analysis"
/>
`;

exports[`Analysis must render correctly nutrition 1`] = `"Nutrition - Analysis"`;

exports[`Analysis must render correctly nutrition: Analysis iframe 1`] = `
<iframe
  className="analysis-iframe"
  frameBorder="0"
  onLoad={[Function]}
  scrolling="yes"
  seamless={true}
  src="https://somesuperseturl.org/superset/dashboard/11/?standalone=true"
  title="pregnancy sms events analysis"
/>
`;

exports[`Analysis must render correctly pregnancy 1`] = `"Pregnancy - Analysis"`;

exports[`Analysis must render correctly pregnancy: Analysis iframe 1`] = `
<iframe
  className="analysis-iframe"
  frameBorder="0"
  onLoad={[Function]}
  scrolling="yes"
  seamless={true}
  src="https://somesuperseturl.org/superset/dashboard/9/?standalone=true"
  title="pregnancy sms events analysis"
/>
`;
